src/Microsoft.AspNetCore.Blazor.Browser/Interop/TrackedReference.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Blazor.Browser.Interop
+{
+    internal class TrackedReference
+    {
+        private TrackedReference(string id, object reference)
+        {
+            Id = id;
+            TrackedInstance = reference;
+        }
+        private static IDictionary<string, TrackedReference> References { get; } =
+            new Dictionary<string, TrackedReference>();
+
+        public string Id { get; }
+
+        public object TrackedInstance { get; }
+
+        public static void Track(string id, object reference)
+        {
+            var trackedRef = new TrackedReference(id, reference);
+            if (References.ContainsKey(id))
+            {
+                throw new InvalidOperationException($"An element with id '{id}' is already being tracked.");
+            }
+
+            References.Add(id, trackedRef);
+        }
+
+        public static void Untrack(string id)
```

Comment:
As with the JS side, looks like this is not yet called. I'm sure it's on your list to implement, but I'm adding this comment for my own benefit to retain brain state across code reviews :)